simpa/utils/tissue_properties.py

```diff
@@ -17,25 +18,9 @@ class TissueProperties(dict):
                      Tags.DATA_FIELD_SPEED_OF_SOUND,
                      Tags.DATA_FIELD_ALPHA_COEFF]
 
-    def __init__(self):
+    def __init__(self, settings: Settings):
```

Comment:
I thought about this a lot, too.

Using it this way the code simplifies from:

    xdim, ydim, zdim = self.global_settings.get_volume_dimensions_voxels()
    props = TissueProperties(xdim, ydim, zdim)

to
    props = TissueProperties(self.global_settings)
    
The global settings dictionary is currently available in every single simulation component anyway, and directly handing over a pointer to it seemed the most convenient, as it also reduces the chance for a user to misassign the x, y, or z coordinates.

One could of course make a point that only the most minimal necessary parameters should be handed over, but I thought in this case there were advantages than risks with the proposed approach.

Reply:
Since python handles objects by reference, I think that this is not an issue. However, I started to wonder if we should make the settings immutable to avoid possible accidental changes during runtime (for another feature of course).